scylla/src/frame/request/query.rs

```diff
@@ -18,15 +18,15 @@ const FLAG_WITH_SERIAL_CONSISTENCY: u8 = 0x10;
 // const FLAG_WITH_NAMES_FOR_VALUES: u8 = 0x40;
 
 pub struct Query<'a> {
-    pub contents: String,
+    pub contents: &'a str,
```

Comment:
@piodul do you remember why we originally put an owned String in there instead of &str? A slice would of course be preferred, unless there was some important use case we had in mind, which I don't remember.

Reply:
I don't know. If lifetimes check out, it can be a `&str`.

Reply:
I think we wanted to be able to write things like:
```rust
Query::from(format!("INSERT INTO {} ...", my_table_name))
```
Where we need to create a `Query` from a `String` that will be dropped in a moment

Maybe a

Reply:
Maybe `Cow<&str>` would work for both use cases

Reply:
We have two `Query` structs - one is user facing, and the latter is used in serialization. I agree that your argument applies to the first one, but the struct in question here is the second one.

Reply:
I think that we used to have a model in which the connection worker received requests formatted as structs and sent back deserialized responses as structs, too. Now, the worker receives serialized requests and sends serialized responses. In the previous model, the struct needed to have a `'static` lifetime in order to be pushed through a channel - perhaps this was the reason this field is a `String`.

Reply:
Maybe T: AsRef<&str> would work, I can check it out

Reply:
> We have two Query structs - one is user facing, and the latter is used in serialization.

right, I looked at the wrong one. `&'a str` seems fine then.

Reply:
I don't really think making the struct generic over the string type is necessary. It's not a user facing struct (although I think it was unnecessarily made public), so if we later need it to be more "elastic", we can make it generic later.

If it compiles now, IMO it's OK to simply use &str.

Reply:
@piodul ah ok, I can take a look at the other struct as well. In the ORM I am writing, a lot of queries are &’static str. It would be a bummer if the user interface query would allocate all those str’s to Strings

Reply:
I guess it would be chaos to introduce a Cow<&str>, AsRef<str> or &'a str. String makes it very easy, let's keep it that way